### ISSUE TITLE
Create skeleton worker to run a build task

### ISSUE THREAD

**Issue by apburnes** [cloud-gov/pages-core#4185]
Create a skeleton build task that runs from the pages worker, updates the build task queue, and saves an artifact.

## Acceptance Criteria

- [ ] Create a build task queue
- [ ] Run build task when a site build starts
- [ ] Run task from worker thread
- [ ] Update tasks queue with new status
- [ ] On task complete, save task artifact and complete tasks queue with finish status.

**Comment by drewbo** [comment]
One question before we begin implementation on this: how does a build know which build task types to kick off? I think the one missing piece from the ADR is that the build table should also have a `task_types` column which is an array of build task types? The feels a little non-ideal (all of `build`, `build_task`, and `build_task_type` would have dependencies on one another) but we need to track which types of tasks to start, assuming that users have control over which tasks they want to opt-in to on each build

**Comment by apburnes** [comment]
Great point! Maybe we add a site `site_build_task` table to manage all of the build tasks that run against the site where we can specify the `build_task_type`, when/what it runs on, etc. Or we change `build_task_type` table to better encompass the site, the configuration, and `task_types`.

**Comment by drewbo** [comment]
I like the first option `site_build_task` and it should probably be branch specific. I definitely have the feeling that we're going to refactor this all in the future when we do a broader "configuration review" but I think this should work for an initial implementation:
```mermaid
erDiagram
    SITE_BUILD_TASK {
        int id PK
        int site FK "site"
        int build_task_type FK "build_task_type"
        string branch
        json metadata
    }
```

**Comment by apburnes** [comment]
The table makes sense to me. `branch` would be nullable if we want the task to run on every build?

**Comment by drewbo** [comment]
Yeah that works, I'm going to include the `site_build_task` table in #4245 before proceeding

**Comment by drewbo** [comment]
Closed via #4248
